backend/routers/services.py

```
@@ -2,67 +2,89 @@
 from db import get_db
 from typing import List, Dict, Any
 import json
+import logging
+
+# Import dynamic comprehensive pricing
+try:
+    from dynamic_comprehensive_pricing import DynamicComprehensivePricing
+    DYNAMIC_PRICING_AVAILABLE = True
+except ImportError:
+    DYNAMIC_PRICING_AVAILABLE = False
+    logger.warning("Dynamic comprehensive pricing not available - falling back to basic pricing")
```

Ruff (0.12.2)

Undefined name `logger`

(F821)